plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStoreMeta.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * OrdersTableDataStoreMeta class file.
+ */
+
+namespace Automattic\WooCommerce\Internal\DataStores\Orders;
+
+/**
+ * Mimics a WP metadata (i.e. add_metadata(), get_metadata() and friends) implementation using a custom table.
+ */
+class OrdersTableDataStoreMeta {
+
+	/**
+	 * Describes the structure of the metadata table.
+	 *
+	 * @return array Array elements: table, object_id_field, meta_id_field.
+	 */
+	protected function get_db_info() {
+		global $wpdb;
+
+		return array(
+			'table'           => OrdersTableDataStore::get_meta_table_name(),
+			'meta_id_field'   => 'id',
+			'object_id_field' => 'order_id',
+		);
+	}
+
+	/**
+	 * Returns an array of meta for an object.
+	 *
+	 * @param  WC_Data $object WC_Data object.
+	 * @return array
+	 */
+	public function read_meta( &$object ) {
+		global $wpdb;
+
+		$db_info = $this->get_db_info();
+
+		// XXX: do we want to augment this with pre-COT metadata?
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$raw_meta_data = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT {$db_info['meta_id_field']} AS meta_id, meta_key, meta_value FROM {$db_info['table']} WHERE {$db_info['object_id_field']} = %d ORDER BY meta_id",
+				$object->get_id()
+			)
+		);
+		// phpcs:enable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+
+		return $raw_meta_data;
+	}
+
+	/**
+	 * Deletes meta based on meta ID.
+	 *
+	 * @param  WC_Data  $object WC_Data object.
+	 * @param  stdClass $meta (containing at least ->id).
+	 */
+	public function delete_meta( &$object, $meta ) {
+		global $wpdb;
+
+		if ( ! is_a( $meta, 'WC_Meta_Data' ) || ! isset( $meta->id ) ) {
+			return false;
+		}
+
+		$db_info = $this->get_db_info();
+
+		$meta_id      = absint( $meta->id );
+		$current_meta = $this->get_metadata_by_id( $meta_id );
```

Comment:
why do we need to fetch the metadata? we can directly fire the delete command and figure out whether it's needed based on value returned by `$wpdb->delete`

Reply:
Done.